dev/ide/profile-workspace.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Run node ./dev/ide/capture-ws-top.js to profile cpu and memory usage of the workspace.
+ *
+ */
+//@ts-check
+
+const { execSync } = require("child_process");
+const { promises } = require("fs");
+const path = require("path");
+
+/**
+ * @param {number[]} arr
+ * @returns {number}
+ */
+const q90 = (arr) => {
+    const sorted = arr.sort((a, b) => a - b);
+    const pos = (sorted.length - 1) * 0.9;
+    const base = Math.floor(pos);
+    const rest = pos - base;
+    if (sorted[base + 1] !== undefined) {
+        return sorted[base] + rest * (sorted[base + 1] - sorted[base]);
+    } else {
+        return sorted[base];
+    }
+};
+
+(async () => {
+    let wsPodName;
+    while (!wsPodName) {
```

Comment:
`wsPodName` only setup once. we need to restart script next time to get correct pod name.

You can ignore it since it's not important